src/test/java/test/priority/issue2137/IssueTest.java

```diff
@@ -0,0 +1,75 @@
+package test.priority.issue2137;
+
+import org.testng.annotations.Test;
+
+public class IssueTest {
+
+  @Test(priority = Integer.MIN_VALUE)
+  public void a() {
+  }
+
+  @Test(priority = 1)
```

Comment:
You should try with more specific values like 0 and negative priorities

Reply:
@juherr - Negative priorities already got covered when I used `Integer.MIN_VALUE`. I didn't use `0` because that would also cause the priorities to be side-lined (when two priorities are equal, TestNG looks for test name as the tie breaker)